tests/test_code_directory.py

```diff
@@ -87,11 +96,18 @@ def test_match_excluded_precedence_over_included(self, dir_structure):
         expected = ["empty_for_testing.py", "insecure_random.py"]
         files = match_files(
             dir_structure,
-            exclude_paths=["*request.py"],
+            exclude_paths=["**/tests/**", "*request.py"],
             include_paths=["*request.py", "*empty*.py", "*random.py"],
         )
         self._assert_expected(files, expected)
 
+    def test_test_directory_not_excluded(self, dir_structure):
```

Comment:
given all the indexing and iteration now in code_directory.py I was hoping to see more unit tests, including times when a user may pass in non-glob compliant patterns. also test a few of the default patterns. Maybe it's already there and I just didn't see the tests, but pointing it out

Reply:
Yeah I relied on the existing tests a lot for this work but I could add some more.

Reply:
@clavedeluna I added a few more tests showing default behavior and override with respect to test directories. I'm not sure what to do about non-compliant patterns; that's basically user error and it just means things aren't going to work they way they expect.

Reply:
Yeah that's fair.